### PR TITLE
Deletes storage on component deletion

### LINKED ISSUE #2529: odo delete does not delete user specified storage
/kind bug


## What versions of software are you using?

**Operating System:**
Win10 Pro

**Output of `odo version`:**
```
odo v1.0.3 (HEAD)

Server: https://172.18.1.78:8443
Kubernetes: v1.11.0+d4cacc0
```

## How did you run odo exactly?
```
odo storage create ...
odo push
odo delete
odo push
```

## Actual behavior

```
Validation                                                                                                                                                                                    
 V  Checking component [52ms]                                                                                                                                                                 
                                                                                                                                                                                              
Configuration changes                                                                                                                                                                         
 X  Failed to create component with name qcomp. Please use `odo config view` to view settings used to create component. Error: unable to create PVC: unable to create PVC: persistentvolumecla
ms "stor1-qapp-pvc" already exists                                                                                                                                                            
```

## Expected behavior

Component should be pushed again without error

## Any logs, error output, etc?

### PR DESCRIPTION
**Pull request by mik-dass**
**What type of PR is this?**

This PR deletes storage belonging to a component on component deletion

/kind bug

**What does does this PR do / why we need it**:

Fixes storage deletion on `odo delete`

**Which issue(s) this PR fixes**:

Fixes #2529 

**How to test changes / Special notes to the reviewer**:

- create a component
- add a storage to it
- delete the component
- push again and check if we face any issues.

### CODE REVIEW

File: pkg/occlient/occlient_test.go [hunk 1]
```diff
@@ -1247,7 +1277,12 @@ func TestNewAppS2I(t *testing.T) {
 					t.Errorf("deploymentconfig name is not matching with expected value, expected: %s, got %s", tt.args.commonObjectMeta.Name, createdDC.Spec.Selector["deploymentconfig"])
 				}
 
-				createdSvc := fkclientset.Kubernetes.Actions()[0].(ktesting.CreateAction).GetObject().(*corev1.Service)
+				var createdSvc *corev1.Service
+				if len(tt.args.storageToBeMounted) > 0 {
+					createdSvc = fkclientset.Kubernetes.Actions()[4].(ktesting.CreateAction).GetObject().(*corev1.Service)
```
Comment (girishramnani) [inline pkg/occlient/occlient_test.go]: can you explain the `[4]` here? I mean can we have a comment here to explain the list of `Actions()`?

Comment (mik-dass) [reply pkg/occlient/occlient_test.go]: @girishramnani fixed

**Pull request merged**